pilot/pkg/xds/ads.go

```diff
@@ -112,6 +113,15 @@ type Connection struct {
 	blockedPushes map[string]*model.PushRequest
 }
 
+// Request is the wrapper of requested xds type or blocked pushRequest
+type Request struct {
+	// pushRequest represents the blocked push request.
+	pushRequest *model.PushRequest
+
+	// resource type
+	typeURL string
+}
+
```

Comment:
This is what i send to the reqChan, it has no reference to DiscoveryRequest any more. Let alone, i could not think of when we did not read the buffered channel.